src/main/java/net/minecraftforge/common/command/SelectorHandlerManager.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common.command;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import org.apache.commons.lang3.ArrayUtils;
+
+import net.minecraft.command.CommandException;
+import net.minecraft.command.EntitySelector;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.entity.Entity;
+import net.minecraftforge.fml.common.FMLLog;
+import net.minecraftforge.fml.common.Loader;
+
+/**
+ * Allows registration of custom selector types by assigning a {@link SelectorHandler} to a prefix
+ * To check whether a new prefix conflicts with any registered prefix, the methods {@link #isShadowed} and {@link #isShadowing} can be used
+ * This class handles calls to the {@link EntitySelector} methods {@link EntitySelector#matchEntities matchEntities},
+ * {@link EntitySelector#matchesMultiplePlayers matchesMultiplePlayers} and {@link EntitySelector#hasArguments hasArguments}.<br>
+ * The calls are delegated to the handler with the longest matching prefix
+ */
+public class SelectorHandlerManager
+{
+    private SelectorHandlerManager()
+    {
+    }
+
+    //the ordering is reversed such that longer prefixes appear before their shorter substrings
+    private static final SortedMap<String, SelectorHandler> selectorHandlers = new TreeMap<String, SelectorHandler>(Collections.<String> reverseOrder());
+    private static final SortedMap<String, String> registeringMods = new TreeMap<String, String>(Collections.<String> reverseOrder());
```

Comment:
Only add one written instead of another stupid library. Then just left it like that is good.